be/src/connector/lake_connector.h

```diff
@@ -0,0 +1,146 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "connector/connector.h"
+#include "exec/vectorized/olap_scan_prepare.h"
+#include "gen_cpp/InternalService_types.h"
+#include "runtime/global_dict/parser.h"
+#include "storage/conjunctive_predicates.h"
+#include "storage/lake/tablet_reader.h"
+
+namespace starrocks::connector {
+
+class LakeConnector final : public Connector {
+public:
+    ~LakeConnector() override = default;
+
+    DataSourceProviderPtr create_data_source_provider(vectorized::ConnectorScanNode* scan_node,
+                                                      const TPlanNode& plan_node) const override;
+};
+
+class LakeDataSource;
+
+class LakeDataSourceProvider final : public DataSourceProvider {
+public:
+    ~LakeDataSourceProvider() override = default;
+    friend class LakeDataSource;
+    LakeDataSourceProvider(vectorized::ConnectorScanNode* scan_node, const TPlanNode& plan_node);
+    DataSourcePtr create_data_source(const TScanRange& scan_range) override;
+
+protected:
+    vectorized::ConnectorScanNode* _scan_node;
+    const TOlapScanNode _t_olap_scan_node;
+};
+
+// TODO: support parallel scan within a single tablet
+class LakeDataSource final : public DataSource {
```

Comment:
Can we move the class definition to cpp file to reduce compile-time dependency.

Reply:
done